File: sdks/redlite-java/src/main/java/com/redlite/GeoMember.java
package com.redlite;

/**
 * Geospatial member with coordinates.
 */
public record GeoMember(double longitude, double latitude, String member) {

    /**
     * Create a GeoMember.
     */
    public static GeoMember of(double longitude, double latitude, String member) {
        return new GeoMember(longitude, latitude, member);
    }
}

File: sdks/redlite-kotlin/src/main/kotlin/com/redlite/GeoMember.kt
package com.redlite

/**
 * Geospatial member with coordinates.
 *
 * @property longitude The longitude coordinate
 * @property latitude The latitude coordinate
 * @property member The member name
 */
data class GeoMember(
    val longitude: Double,
    val latitude: Double,
    val member: String
) {
    companion object {
        @JvmStatic
        fun of(longitude: Double, latitude: Double, member: String) =
            GeoMember(longitude, latitude, member)
    }
}
